rfcs/text/0408-versioning-types.md

```diff
@@ -0,0 +1,344 @@
+- Feature Name: versioning-types
+- Start Date: 2022-06-28
+- RFC PR: [blockprotocol/blockprotocol#408](https://github.com/blockprotocol/blockprotocol/pull/408)
+- Block Protocol Discussion: [blockprotocol/blockprotocol#408](https://github.com/blockprotocol/blockprotocol/discussions/408)
+
+# Summary
+
+[summary]: #summary
+
+The Type System outlined in [RFC 0352](https://github.com/blockprotocol/blockprotocol/blob/am/rfc-graph-type-system/rfcs/text/0352-graph-type-system.md) introduced classes of types that are used to describe the data used and produced by blocks and embedding applications. The types are possibly (ideally often) publicly accessible, and their hierarchy allows for them to be composed together into new types. Due to this, changes in a type can affect the structure of another one. This RFC defines how types are versioned, to mitigate potential problems that would arise without versioning.
+
+# Motivation
+
+[motivation]: #motivation
+
+The Type System is intended to allow for **reusable and shared** descriptions of data. As such, a version of a block should be able to tie itself to a _specific_ description of the structure of the data it requires as if types are able to be updated independently of blocks then this would cause problems. Additionally, if someone (Alice) creates an entity type that refers to someone else's (Bob's) "address" property type, we do not want Bob to be able to change what Alice meant in her entity type by arbitrarily and implicitly updating it.
```

Comment:
Sorry yeah I missed another clause, I was trying to say if Blocks _can't_ pin themselves to specific versions _and_ types can be updated independently (which we want), then we encounter problems.

I like your wording, thank you. Although I'm going to omit "as one of its properties" because it's sort of redundant and I _try_ and avoid "property" on its own when talking about Types because we enter zones of ambiguity of if we're talking about JSON stuff properties, property _types_, properties in JSON objects, keys or values, etc.

Reply:
Sure, sounds good